File: DotNet/BIAPackage/BIA.Net.Core.Presentation.Api/HttpErrorReport.cs
// <copyright file="HttpErrorReport.cs" company="PlaceholderCompany">
// Copyright (c) PlaceholderCompany. All rights reserved.
// </copyright>

namespace BIA.Net.Core.Presentation.Api
{
    /// <summary>
    /// Represents a HTTP error report.
    /// </summary>
    /// <param name="ErrorCode">The error code.</param>
    /// <param name="ErrorMessage">The error message.</param>
    public record class HttpErrorReport(int ErrorCode,  string ErrorMessage)
    {
    }
}

File: Angular/packages/bia-ng/models/http-error-report.ts
export interface HttpErrorReport {
  errorCode: number;
  errorMessage: string;
}

export function isHttpErrorReport(error: any): error is HttpErrorReport {
  return (
    typeof error === 'object' &&
    error !== null &&
    typeof error.errorCode === 'number' &&
    typeof error.errorMessage === 'string'
  );
}
